test/distributed/_tensor/experimental/test_tensor_parallel.py

```diff
@@ -0,0 +1,61 @@
+# Owner(s): ["oncall: distributed"]
+import torch
+from torch.distributed._tensor.experimental.tensor_parallel import (
+    tensor_parallel_transformation,
+)
+from torch.testing._internal.distributed._tensor.common_dtensor import (
+    DTensorTestBase,
+    with_comms,
+)
+
+
+class MLPListModule(torch.nn.Module):
+    """
+    A dummy model with list of MLPs.
+    """
+
+    def __init__(self, num_mlps=3):
+        super().__init__()
+        self.mlps = torch.nn.ModuleList()
+        for _ in range(num_mlps):
+            self.mlps.append(
+                torch.nn.Sequential(
+                    torch.nn.Linear(6, 8),
+                    torch.nn.ReLU(),
+                    torch.nn.Linear(8, 6),
+                )
+            )
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x = torch.chunk(x, 2, dim=1)[0]
+        for mlp in self.mlps:
+            x = mlp(x)
+        return x + torch.ones_like(x)
+
+
+class TensorParallelTest(DTensorTestBase):
+    def setUp(self) -> None:
+        super().setUp()
+
+    @with_comms
+    def test_tensor_parallel_transformation_e2e(self):
+        torch.manual_seed(0)
+        model = MLPListModule(2).to(device=self.device_type)
+        inputs = (torch.randn((10, 12)).to(device=self.device_type),)
+        with torch.inference_mode():
```

Comment:
Currently this is pretty commonly used in model processing and inference since gradient calculation is not needed. I'm curious why some aten ops may not decompose under inference mode. Is this something WIP or by design?

Reply:
Something it's a bit weird under torch.inference_mode(), for example, if you capture ops inside inference_mode, it would give you either `aten.linear.default` or `aten.matmul.default`, instead of giving you `aten.mm/addmm.default` instead. That is probably ok if torch.export is not run under `inference_mode`

Reply:
Interesting, that's good to know.